docs/INTERNAL_METRICS.md

```diff
@@ -0,0 +1,52 @@
+# LogDNA Agent Internal Metrics
+
+The LogDNA agent records metrics that can be relevant for monitoring and alerting, such as number log files currently
+tracked or number of bytes parsed, along with process status information.
+
+## Exposing the Agent Metrics as a Prometheus Endpoint 
+
+You can enable the agent's [Prometheus][prometheus] endpoint to expose the internal metrics to a Prometheus server,
+by setting the `LOGDNA_METRICS_PORT` environment variable to an available port number, for example:
+
+```yaml
+    - env:
+        - name: LOGDNA_METRICS_PORT
+          value: "9881"
+```
+
+Metrics related to the agent are exposed using the prefix `logdna_` and process status information, like memory and
+cpu usage, are exposed with the prefix `process_`.
+
+## Enabling Prometheus target discovery on Kubernetes
```

Comment:
It may be worth adding something about PodMonitors iff the customer is running the prometheus operator (they might be confused and try to use a ServiceMonitor)

https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/api.md#podmonitor

https://github.com/prometheus-operator/prometheus-operator/blob/master/example/user-guides/getting-started/example-app-pod-monitor.yaml

Reply:
I think the default installation of prometheus and vendors have discovery based on `prometheus.io/port` and `prometheus.io/scrape` enabled by default (I've tried vanilla prometheus server and sysdig), so we hopefully we can avoid documenting here prometheus server settings :)